### PR TITLE
[feat]curvefs: support memcached in container

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
1. add support in entrypoint.sh
2. add memcached install in bash image
3. copy dir by module when making image, which increases the number of layers of the image, but reduces the burden of uploading

Signed-off-by: Cyber-SiKu <Cyber-SiKu@outlook.com>

### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

File: curvefs/docker/debian10/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,15 @@
 FROM opencurvedocker/curve-base:debian10
 ENV TZ=Asia/Shanghai
 RUN mkdir -p /curvefs /etc/curvefs /core /etc/curve
-COPY curvefs /curvefs
+COPY curvefs/client /curvefs
```
Comment (Wine93) [inline curvefs/docker/debian10/Dockerfile:4]: I think only one OS Dockerfile needs to be kept for running curve in docker (maybe debian9 or debian10), it's too many to maintain.

File: curvefs/docker/debian10/entrypoint.sh [hunk 2]
```diff
@@ -87,6 +87,14 @@ function prepare() {
             g_binary="$g_prefix/sbin/curve-fuse"
             g_start_args="--confPath $conf_path"
             ;;
+        monitor)
+            g_binary="python3"
+            g_start_args="target_json.py"
+            ;;
+        memcached)
+            g_prefix="/memcached"
```
Comment (Wine93) [inline curvefs/docker/debian10/entrypoint.sh:95]: Whether memcache is part of the curvefs system (like snapshot clone server)? If so, i think place it to `/curvefs` directory is better.

File: curvefs/docker/debian10/Dockerfile [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+COPY curvefs/mds /curvefs
+COPY curvefs/metaserver /curvefs
+COPY curvefs/monitor /curvefs
+COPY curvefs/tools /curvefs
 COPY entrypoint.sh /
 COPY curvefs/tools/sbin/curvefs_tool /usr/bin
-COPY libmemcached.so /usr/lib/
-COPY libmemcached.so.11 /usr/lib/
-COPY libhashkit.so.2 /usr/lib
+COPY libmemcached.so libmemcached.so.11 libhashkit.so.2 /usr/lib/
```
Comment (Wine93) [inline curvefs/docker/debian10/Dockerfile:13]: 👍

File: curvefs/docker/debian10/Dockerfile [hunk 4]
```diff
@@ -1,11 +1,15 @@
 FROM opencurvedocker/curve-base:debian10
 ENV TZ=Asia/Shanghai
 RUN mkdir -p /curvefs /etc/curvefs /core /etc/curve
-COPY curvefs /curvefs
+COPY curvefs/client /curvefs
+COPY curvefs/conf /curvefs
```
Comment (Wine93) [inline curvefs/docker/debian10/Dockerfile:5]: Why use multi `COPY` commands to copy curvefs directory?

Comment (Cyber-SiKu) [reply curvefs/docker/debian10/Dockerfile:5]: > Why use multi `COPY` commands to copy curvefs directory?

Most of the time, only one or two modules in client, mds or metaserver may be updated, but the entire /curvefs directory needs to be updated, so the entire directory needs to be updated when pushing or pulling.

File: curvefs/docker/debian10/Dockerfile (hunk 1 above)
Comment (Cyber-SiKu) [reply curvefs/docker/debian10/Dockerfile:4]: > I think only one OS Dockerfile needs to be kept for running curve in docker (maybe debian9 or debian10), it's too many to maintain.

removed

File: curvefs/docker/debian10/entrypoint.sh (hunk 2 above)
Comment (Cyber-SiKu) [reply curvefs/docker/debian10/entrypoint.sh]: > Whether memcache is part of the curvefs system (like snapshot clone server)? If so, i think place it to `/curvefs` directory is better.

Will memcached be used by other systems in the future? Isn't this bound to curvefs?

**Comment by Cyber-SiKu** [comment]
cicheck

**Review by wu-hanqing — APPROVED** [review]

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Review by Wine93 — APPROVED** [review]

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Pull request merged**